lib/pos_epoch_complete_hook.go

```diff
@@ -121,19 +121,26 @@ func (bav *UtxoView) runEpochCompleteSnapshotGeneration(epochNumber uint64) erro
 	// Snapshot the current GlobalParamsEntry.
 	bav._setSnapshotGlobalParamsEntry(bav.GetCurrentGlobalParamsEntry(), epochNumber)
 
-	// Snapshot the current top m validators as the validator set.
-	validatorSet, err := bav.generateAndSnapshotValidatorSet(epochNumber)
+	currentGlobalParamsEntry := bav.GetCurrentGlobalParamsEntry()
+	maxValidatorsToFetch := currentGlobalParamsEntry.ValidatorSetMaxNumValidators
```

Comment:
nit: please keep naming consistent so it's easier to trace
```suggestion
	currentGlobalParamsEntry := bav.GetCurrentGlobalParamsEntry()
	validatorSetMaxNumValidators := currentGlobalParamsEntry.ValidatorSetMaxNumValidators
	
```